core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -96,6 +109,30 @@ private function addMissingTablesNeededForJoins()
         }
     }
 
+    private function addMissingTablesForOtherTableJoin($tableName, $dependentTable)
+    {
+        $this->tables->addTableDependency($dependentTable, $tableName);
+
+        if ($this->tables->hasJoinedTable($tableName)) {
+            return;
+        }
+
+        $table = $this->tables->getLogTable($tableName);
+
+        if ($table->getColumnToJoinOnIdAction() || $table->getColumnToJoinOnIdAction() || $table->getLinkTableToBeAbleToJoinOnVisit()) {
+            $this->tables->addTableToJoin($tableName);
+            return;
+        }
+
+        $otherTableJoins = $table->getWaysToJoinToOtherLogTables();
+
+        foreach ($otherTableJoins as $logTable => $column) {
+            $this->addMissingTablesForOtherTableJoin($logTable, $tableName);
```

Comment:
Feel free to ignore but in theory we would maybe only add the tables actually needed. Eg if there's `log_form_foo` and it can be joined with `log_form_bar` and `log_form_baz`, then we would maybe only add `log_form_bar` if this leads us to another log table that we can resolve? It's bit edge case and complicated to solve... the same way as figuring out if a table can be resolved or not... it's a rare feature and so far there's always only one dependency I think so it can be ignored probably until maybe needed or when there's a problem.

Reply:
So you mean if `getWaysToJoinToOtherLogTables` defines multiple ways to join to different tables?
I'm currently kind of expecting that there is only one possibility defined.